Source/Parallelization/GuardCellManager.cpp

```diff
@@ -121,8 +123,22 @@ guardCellManager::Init (
     {
         for (int i = 0; i < AMREX_SPACEDIM; i++)
         {
-            ng_alloc_Rho[i] += static_cast<int>(std::ceil(PhysConst::c * dt / dx[i]));
-            ng_alloc_J[i]   += static_cast<int>(std::ceil(PhysConst::c * dt / dx[i] * 0.5_rt));
+            amrex::Real dt_Rho = dt;
+            amrex::Real dt_J = 0.5_rt*dt;
+            if (do_multi_J) {
+                // With multi_J + time averaging, particles can move during 2*dt per PIC cycle.
+                if (fft_do_time_averaging){
+                    dt_Rho = 2._rt*dt;
+                    dt_J = 2._rt*dt;
+                }
+                // With multi_J but without time averaging, particles can move during dt per PIC
+                // cycle for the current deposition as well.
+                else {
+                    dt_J = dt;
+                }
```

Comment:
I'm thinking that maybe we could shorten this a little bit by using the conditional operator:
```
dt_Rho = (fft_do_time_averaging) ? 2._rt*dt : dt_Rho;
dt_J   = (fft_do_time_averaging) ? 2._rt*dt : dt;
```